bundles/smartgrid.newsimcontrol.rcp/src/smartgrid/newsimcontrol/application/commands/ControllerCommand.java

```diff
@@ -0,0 +1,50 @@
+package smartgrid.newsimcontrol.application.commands;
+
+import java.io.FileInputStream;
+import java.io.ObjectInputStream;
+import java.rmi.RemoteException;
+
+import couplingToICT.SimcontrolException;
+import smartgrid.newsimcontrol.controller.LocalController;
+
+public abstract class ControllerCommand {
+
+	LocalController controller; 
+	
+	public ControllerCommand(LocalController controller) {
+		this.controller = controller;
+	}
+	
+	public Object execute(String[] args) throws SimcontrolException, RemoteException, InterruptedException {
+		if (checkArguments(args))
+			return doCommand(args);
+		else 
+			return null;
+	}
+	
+	public abstract boolean allow();
+	
+	public abstract boolean checkArguments(String[] args);
+	
+	public abstract Object doCommand(String[] args) throws RemoteException, SimcontrolException, InterruptedException;
```

Comment:
I'm not entirely sure, why it is necessary to throw RemoteExceptions. We should try to avoid them in the local part and only use them in the remote interface